signer/src/bitcoin/utxo.rs

```diff
@@ -2743,28 +2787,28 @@ mod tests {
         // Create deposit and withdrawal requests, some with too low of a
         // max fees and some with a good max fee.
         let deposit_low_fee = ((SOLO_DEPOSIT_TX_VSIZE - 1.0) * fee_rate) as u64;
-        let low_fee_deposits = std::iter::repeat_with(|| uniform.sample(&mut OsRng))
+        let low_fee_deposits = std::iter::repeat_with(|| uniform.sample(&mut rng.clone()))
```

Comment:
Note that `.clone()` here (and everywhere else) may not have the behavior you might expect; it doesn't behave like a `Rc<Rng>` (i.e. shared ownership), every `clone()` is giving you an exact clone of `rng`:
https://play.rust-lang.org/?version=stable&mode=debug&edition=2024&gist=d0538e45a3c396db1ba714e5af6c9105

Reply:
I don't know exactly but I believe probability of any problems because of such clones are pretty small

Reply:
Whoops, actually such clones _are_ a problem, will remove them

Reply:
Had a conversation about this online and this can be an issue because in some of our tests the randomness is used during key generation and such.